imagetest/test_suites/network/gvnic_test.go

```diff
@@ -0,0 +1,54 @@
+//go:build cit
+// +build cit
+
+package network
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const (
+	driverInfoPath = "/sys/class/net/%s/device/uevent"
+)
+
+var (
+	driverRegex = regexp.MustCompile("DRIVER=[a-zA-Z0-9_-]+")
+)
+
+func CheckGVNICPresent(interfaceName string) error {
+	file := fmt.Sprintf(driverInfoPath, interfaceName)
+	_, err := os.Stat(file)
+	if os.IsNotExist(err) {
+		return err
+	}
+	fileData, err := os.ReadFile(file)
+	if err != nil {
+		return err
+	}
+	driver := driverRegex.FindString(string(fileData))
```

Comment:
this is a very interesting approach, but i think it is easier to do using link resolution. the file /sys/class/net/$device/device/driver/module will be a symbolic link to the driver - this can be looked up using `os.Readlink()` without ever reading any file

Reply:
yes agreed `os.Readlink()` would be a more easier approach.